[CLOSES] Issue#2431 Term Entry: JavaScript Dates .setMilliseconds() (#2615)
Codecademy/docs: content/javascript/concepts/dates/terms/setMilliseconds/setMilliseconds.md

---
Title: '.setMilliseconds()'
Description: 'Sets the milliseconds of a date.'
Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Date'
  - 'Methods'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

The **`.setMilliseconds()`** method sets the milliseconds part of a Date object to desired value. The `.setMilliseconds()` takes an integer argument ranging from 0 to 999, representing the new milliseconds value to be set. Only the milliseconds part of a date and time is modified. 

## Syntax

```pseudo
dateObject.setMilliseconds(n)
```

> Above code changes millisecond of `dateObject` to n, which is an integer between 1 and 999.

Reviewer: ```suggestion
Above code changes millisecond of `dateObject` to `n`, which is an integer between 1 and 999.
```

The  return value of `.setMilliseconds()` is the new timestamp after updating milliseconds value of the original Date object in place. The `.setMilliseconds()` method returns the new timestamp value, which represents the number of milliseconds since January 1, 1970 00:00:00 UTC.

## Example

> To set the millisecond value of given date to 421 :

Reviewer 2: ```suggestion
To set the millisecond value of given date to 421:
```

```js
const eventDate = new Date('1/1/2023');
console.log("Original date:", eventDate);

eventDate.setMilliseconds(421);
console.log("Modified date:", eventDate);
```

This example results in the following output:

```shell
Original date: 2023-01-01T00:00:00.000Z
Modified date: 2023-01-01T00:00:00.421Z
```

> ## Codebyte Example

Reviewer: This codebyte is exactly the same as the example. Can we rewrite it so it's different, at least by using the current date.

> The example below is runnable, and first, it defines `eventDate` with given date and time, then the millisecond value of `eventDate` is changed to 112 milliseconds in future. Lastly, a `console.log` call prints the new value of `eventDate` to the screen:

Reviewer 2: ```suggestion
The example below is runnable and demonstrates how to define `eventDate` with a given date and time. It then modifies the millisecond value of `eventDate` to be 112 milliseconds in the future. Finally, a `console.log` call prints the updated value of `eventDate` to the screen:
```
